### PR TITLE
 Make lib.rs the source of documentation truth.

### PR DESCRIPTION
**Pull request by ltratt**
This takes a suggestion from Edd Barrett, and gives up keeping `README.md` and `src/lib.rs` in sync: `README.md` is now a subset of the documentation in `src/lib.rs`, where the latter is the source of truth.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -33,15 +33,14 @@
 //!                 .skip_while(|l| !l.starts_with(COMMENT_PREFIX))
 //!                 // Extract consecutive commented lines.
 //!                 .take_while(|l| l.starts_with(COMMENT_PREFIX))
-//!                 // Strip the initial "//" from commented lines.
```
Comment (vext01) [inline src/lib.rs:36]: Did we mean to remove this line?

Comment (ltratt) [reply src/lib.rs:36]: Yeah I've more-or-less just moved README.md into lib.rs.

**Comment by vext01** [comment]
In that case LGTM.

bors r+

**Comment by vext01** [comment]
Is that some doctests failing?

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
@vext01 I think this now ready for squashing.

**Comment by vext01** [comment]
Please squash

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**